mayastor/src/bdev/uring_util.rs

```diff
@@ -0,0 +1,89 @@
+use std::{
+    fs::{File, OpenOptions},
+    io::{BufRead, BufReader, ErrorKind},
+    os::unix::fs::OpenOptionsExt,
+};
+
+pub fn fs_supports_direct_io(path: &str) -> bool {
+    // SPDK uring bdev uses IORING_SETUP_IOPOLL which is usable only on a file
+    // descriptor opened with O_DIRECT. The file system or block device must
+    // also support polling.
+    // This works on at least XFS filesystems
+    match OpenOptions::new()
+        .read(true)
+        .write(true)
+        .custom_flags(libc::O_DIRECT)
+        .open(path)
+    {
+        Ok(_f) => true,
+        Err(e) => {
+            assert_eq!(e.kind(), ErrorKind::InvalidInput);
+            println!("Skipping uring bdev, open: {:?}", e);
+            false
+        }
+    }
+}
+
+fn get_mount_filesystem(path: &str) -> Option<String> {
+    let mut path = std::path::Path::new(path);
+    loop {
+        let f = match File::open("/etc/mtab") {
+            Ok(f) => f,
+            Err(e) => {
+                eprintln!("open: {}", e);
```

Comment:
I'm not sure about these println's and eprintln's in this file. Currently, this library module is used only by a small command -line utility used in our tests to determine if io_uring tests should or should not be run. From that POV that seems to be ok. However, eventually (correct me if I'm wrong) this code is going to be used from mayastor itself to determine if aio or uring should be used for a storage pool. In that case, println's must be removed or replaced by proper log function calls. What is the plan here?

Reply:
Yes, when it is eventually called from the pool code then it should switch to proper logging.